packages/nimble-components/src/rich-text-mention/base/index.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    attr,
+    Notifier,
+    Observable,
+    observable,
+    Subscriber
+} from '@microsoft/fast-element';
+import { FoundationElement } from '@microsoft/fast-foundation';
+import type { MappingConfig } from './models/mapping-config';
+import type {
+    RichTextMentionValidator,
+    RichTextMentionValidity
+} from './models/mention-validator';
+import {
+    MentionInternals,
+    MentionInternalsOptions
+} from './models/mention-internals';
+import { Mapping } from '../../mapping/base';
+
+export type MappingConfigs = ReadonlyMap<string, MappingConfig>;
+
+export interface RichTextMentionConfig {
+    mappingConfigs: MappingConfigs;
+}
+
+/**
+ * The base class for Mention mapping configuration
+ */
+export abstract class RichTextMention<
+    TMentionConfig extends RichTextMentionConfig,
+    TValidator extends RichTextMentionValidator<[]>
+>
+    extends FoundationElement
+    implements Subscriber {
+    public readonly mentionInternals: MentionInternals<TMentionConfig> = new MentionInternals(this.getMentionInternalsOptions());
+
+    @attr
+    public pattern?: string;
+
+    /** @internal */
+    public validator = this.createValidator();
```

Comment:
Mark it readonly and move it up so it is defined right after mentionInternals is defined.

Reply:
Updated